hydrolib/core/io/inifield/models.py

```diff
@@ -0,0 +1,244 @@
+
+
+class AbstractSpatialField(INIBasedModel, ABC):
+    """
+    Abstract base class for `[Initial]` and `[Parameter]` block data in
+    inifield files.
+
+    Defines all common fields. Used via subclasses InitialField and ParameterField.
+    """
+
+    class Comments(INIBasedModel.Comments):
+        quantity: Optional[str] = Field(
+            "Name of the quantity. See UM Table D.2.", alias="quantity"
+        )
+        datafile: Optional[str] = Field(
+            "Name of file containing field data values.", alias="dataFile"
+        )
+        datafiletype: Optional[str] = Field("Type of dataFile.", alias="dataFileType")
+        interpolationmethod: Optional[str] = Field(
+            "Type of (spatial) interpolation.", alias="interpolationmethod"
+        )
+        operand: Optional[str] = Field(
+            "How this data is combined with previous data for the same quantity (if any).",
+            alias="operand",
+        )
+        averagingtype: Optional[str] = Field(
+            "Type of averaging, if interpolationMethod=averaging .",
+            alias="averagingtype",
+        )
+        averagingrelsize: Optional[str] = Field(
+            "Relative search cell size for averaging.", alias="averagingrelsize"
+        )
+        averagingnummin: Optional[str] = Field(
+            "Minimum number of points in averaging. Must be ≥ 1.",
+            alias="averagingnummin",
+        )
+        averagingpercentile: Optional[str] = Field(
+            "Percentile value for which data values to include in averaging. 0.0 means off.",
+            alias="averagingpercentile",
+        )
+        extrapolationmethod: Optional[str] = Field(
+            "Option for (spatial) extrapolation.", alias="extrapolationmethod"
+        )
+        locationtype: Optional[str] = Field(
+            "Target location of interpolation.", alias="locationtype"
+        )
+        value: Optional[str] = Field(
+            "Only for dataFileType=polygon. The constant value to be set inside for all model points inside the polygon."
+        )
+
+    comments: Comments = Comments()
+
+    quantity: str = Field(alias="quantity")
+    datafile: Path = Field(alias="dataFile")
+    datafiletype: DataFileType = Field(alias="dataFileType")
+    interpolationmethod: Optional[InterpolationMethod] = Field(
+        alias="interpolationMethod"
+    )
+    operand: Optional[Operand] = Field(Operand.override, alias="operand")
+    averagingtype: Optional[AveragingType] = Field(
+        AveragingType.mean, alias="averagingType"
+    )
+    averagingrelsize: Optional[NonNegativeFloat] = Field(1.01, alias="averagingRelSize")
+    averagingnummin: Optional[PositiveInt] = Field(1, alias="averagingNumMin")
+    averagingpercentile: Optional[NonNegativeFloat] = Field(
+        0, alias="averagingPercentile"
+    )
+    extrapolationmethod: Optional[bool] = Field(False, alias="extrapolationMethod")
```

Comment:
I do not think this will go correctly without validator unfortunately, because possible values for `extrapolationMethod` are "yes" and "no", instead of "true" and "false" or "0" and "1"

Reply:
I just found out a bug in the kernel here, so I will have that fixed first, and have opened a followup issue here: #186.